### PR TITLE
Ft/s3 c 1115/gcp s3 common

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Common file changes to add GCP backend

### CODE REVIEW

**Comment by alexanderchan-scality** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=test/zenkobb
SCALITY_FEDERATION_BRANCH=ft/S3C-1114/GCP-BACKEND
SCALITY_S3_BRANCH=ft/S3C-1115/GCP-S3-Common
try

File: lib/data/external/utils.js [hunk 1]
```diff
@@ -20,14 +20,14 @@ const backendHealth = {
...
+    logHelper(log, level, description, error, dataStoreName, backendType) {
         log[level](description, { error: error.message,
-          errorName: error.name, dataStoreName });
+          errorName: error.name, dataStoreName, backendType });
     },
     // take off the 'x-amz-meta-'
     trimXMetaPrefix(obj) {
         const newObj = {};
-        Object.keys(obj).forEach(key => {
+        Object.keys(obj || {}).forEach(key => {
```
Comment (bennettbuchanan) [inline lib/data/external/utils.js]: This is a bit hard to follow. Perhaps this could be made more explicit by adding a separate condition for the case which this addresses.

File: lib/data/external/AwsClient.js [hunk 2]
```diff
@@ -39,16 +39,16 @@ class AwsClient {
         const putCb = (err, data) => {
```
Comment (bennettbuchanan) [inline lib/data/external/AwsClient.js]: One more change of this nature is needed to remove "AWS" in the logs here: https://github.com/scality/S3/pull/1130/commits/80ab22592c97f3e5f9cd35e98ebf39ac884d2463#diff-bdedeaa43c674dd6fc4de3fdd0d030bcR109.

**Comment by alexanderchan-scality** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=test/zenkobb
SCALITY_FEDERATION_BRANCH=ft/S3C-1114/GCP-BACKEND
SCALITY_S3_BRANCH=ft/S3C-1115/GCP-S3-Common
try

**Comment by alexanderchan-scality** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=test/zenkobb
SCALITY_FEDERATION_BRANCH=ft/S3C-1114/GCP-BACKEND
SCALITY_S3_BRANCH=ft/S3C-1115/GCP-S3-Common
try
